Gems/AWSCore/cdk/example/example_resources_stack.py

```diff
@@ -42,29 +41,96 @@ def __init__(self, scope: core.Construct, id_: str, project_name: str, feature_n
         self.__create_outputs()
 
         # Finally grant cross stack references
-        self.__grant_access(props=props_)
+        self.__grant_access()
 
-    def __grant_access(self, props: CoreStackProperties):
-        self._s3_bucket.grant_read(props.user_group)
-        self._s3_bucket.grant_read(props.admin_group)
+    def __grant_access(self):
+        user_group = iam.Group.from_group_arn(
+            self,
+            f'{self._project_name}-{self._feature_name}-ImportedUserGroup',
+            core.Fn.import_value(f'{self._project_name}:UserGroup')
+        )
+        admin_group = iam.Group.from_group_arn(
+            self,
+            f'{self._project_name}-{self._feature_name}-ImportedAdminGroup',
+            core.Fn.import_value(f'{self._project_name}:AdminGroup')
+        )
```

Comment:
As you commented in details, core stack should be deployed before core example stack, where is the change to guarantee it? Or cdk will guarantee it automatically as example s3 creation depends on core stack s3?

Reply:
For the core gem, currently we're using the "cdk deploy --all" command to deploy both stacks (Core and CoreExample). With this changes, users should do the deployment using the exact same command and CDK guarantees that the Core stack is deployed before the CoreExample stack. Deployment will fail if users try to deploy the CoreExample stack without the Core stack.

Reply:
This is also part of the problem I want to get addressed at https://jira.agscollab.com/browse/LYN-5149. We don't have any mechanism to make sure that the Core (Common) stack is deployed before users deploy the CDK application for any specific feature.

Reply:
actually you can add dependency to stack, stack.add_dependency(stack), like coreexamplestack.add_dependency(corestack)

Reply:
I believe this dependency has already been added automatically if you deploy these two stacks together. Even we have this line added, users can still deploy the example stack using the standard CDK command like below without deploying the core stack:

- cdk deploy CoreExampleStack